bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/HueBridge.java

```diff
@@ -438,17 +464,16 @@ public Group getAllGroup() {
     public List<FullGroup> getGroups() throws IOException, ApiException {
```

Comment:
ConfigurationException, CommunicationException is missing here (for consistency).

Reply:
Added.

Reply:
But method updateGroups in is calling this method and so should also be marked as throwing `ConfigurationException and, `CommunicationException `.
And then lightPollingRunnable...

If you search a little, you will find that these exceptions should be mentioned at more places.

Reply:
You are right. This seems to be an actual bug since these exceptions are not handled anywhere. So in case of time-out, invalid certificate, etc. the binding might stop working.

Reply:
@cweitkamp - since we didn't make it for 3.3, this comment should probably be looked into.

Reply:
As clarified by [this comment](https://github.com/openhab/openhab-addons/pull/11842#issuecomment-1259970498) the exceptions are handled. So only the consistency matter is left. We could either remove all the explicit declarations of these unchecked exceptions, or add them everywhere for consistency. I would opt for the first option. However, it's only a matter of style and it doesn't make any difference runtime, so I would approve this PR even without changing anything.